Source/WTF/wtf/text/cf/StringImplCF.cpp

```diff
@@ -55,46 +59,38 @@ namespace StringWrapperCFAllocator {
         return CFSTR("WTF::String-based allocator");
     }
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
     static void* allocate(CFIndex size, CFOptionFlags, void*)
     {
         StringImpl* underlyingString = nullptr;
         if (isMainThread()) {
             underlyingString = currentString;
-            if (underlyingString) {
+            if (underlyingString)
                 currentString = nullptr;
-                underlyingString->ref(); // Balanced by call to deref in deallocate below.
```

Comment:
How come we can delete this `ref()`? I can't really tell why.

Reply:
The manual ref and deref now happen automatically when we construct / destruct StringImplWrapper, since it has a RefPtr data member.

Reply:
I'm uploading another version to try to make the RefPtr behavior even clearer.

Reply:
ok